### PR TITLE
Support Unmanaged in Arc

### LINKED ISSUE #2378: Provide ArC version of javax.enterprise.inject.spi.Unmanaged
ArC does not fully support some parts of the CDI SPI such as `AnnotatedType`, `InjectionTarget`. However, we should be able to [inject unmanaged instances](https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcTestResourceProvider.java#L21-L54). We should provide an ArC version of `Unmanaged`.

### PR DESCRIPTION
**Pull request by dufoli**
fix #2378

### CODE REVIEW

**Comment by gsmet** [comment]
@dufoli please make the PR title describe what this is about. Also please make your commit comments meaningful. Thanks.

**Comment by dufoli** [comment]
@gsmet I will squash and rename all commit when it start to work.

**Comment by manovotn** [comment]
I'll take a look tomorrow; need to get a hang of the code and what needs doing.

**Comment by dufoli** [comment]
I do not need a real review, it is more to be allign about the analyze I do in comment https://github.com/quarkusio/quarkus/pull/19802#issuecomment-910412835

**Comment by Ladicek** [comment]
The proposal in https://github.com/quarkusio/quarkus/pull/19802#issuecomment-910412835 would allow using `Unmanaged` for classes that are already beans, which is useless in my opinion. To be able to use `Unmanaged` for classes that are not beans, you'd have to generate "something like bean classes" (intentionally handwavy) for them, and for that, you'd have to have a way to identify which classes should be subject to this.

In other words, I believe that `Unmanaged` can only possibly be made to work some small set of classes, identified at build time (e.g. using an annotation).

To be completely blunt, I'm not even sure we want or need `Unmanaged` supported in Quarkus.

**Comment by manovotn** [comment]
Thinking about this some more and having discussed it with Ladislav, I also tend to think we don't actually want to implement this feature.

The reason being, it doesn't fit into the build-time approach we have - `Unmanaged` is normally used very dynamically and for classes that users doesn't have control over (else you could just turn them into beans). This won't work in Quarkus as you'd need to have an upfront information on what classes are to be used as `Unmanaged` beans (to be able to generate the code for their injection targets) which sort of defeats its purpose.

Furthermore, CDI currently works on a `CDI Lite` variant which is tailored for build-time approach and leaves out some of its features - `Unmanaged` being one of them.

**Comment by dufoli** [comment]
I was thinking that it can be usefull for bean class only maybe.
Anyway I have start the PR so I push the code on the branch.
It is not finished but it compile and beans get new function to do not let arc managed produce, inject, postConstruct, dispose...

So bean can be modified between each step. I do not know if it is usefull but this issue was one of the oldest and was pinned so I take it ...

Can the backlog be cleaned of uneeded issue. Because I do the same with CXF and I get rejected from main repo to quarkiverse. I understand the point but it is little bit frustrating to start something from backlog for nothing. Else can you point me to a really needed item I can work on and be sure that it will be merged if I successfully do it ? thanks

keep the good work :-)

**Comment by manovotn** [comment]
It is not always easy to tell if an issue is needed or otherwise. This issue was created back when we had no idea if it makes sense to implement and no big research/test was done on it; it was merely an idea to explore. Some time has passed since, work has been done on CDI Lite and your PR showed up and thanks to all of that we can now see that it doesn't really make sense to support it as it would be pretty limited in scope and not functionally equivalent to what you can achieve with CDI in standard EE containers.

Closing this PR as we decided against implementing this feature for the time being.

@dufoli if you are looking for another issue where you could contribute, I can suggest for instance https://github.com/quarkusio/quarkus/issues/11942 (with [this comment](https://github.com/quarkusio/quarkus/issues/11942#issuecomment-912542734) explaining which part is missing) or https://github.com/quarkusio/quarkus/issues/18982.
